const HoistIntoGlobalsPass:registerDependentDialectsFn  miss initiaization

### What happened?

```
[6964/8543] Building CXX object compiler/src/iree/compiler/Dialect/Util/Transforms/CMakeFiles/iree_compiler_Dialect_Util_Transforms_Transforms.objects.dir/HoistIntoGlobals.cpp.o
FAILED: compiler/src/iree/compiler/Dialect/Util/Transforms/CMakeFiles/iree_compiler_Dialect_Util_Transforms_Transforms.objects.dir/HoistIntoGlobals.cpp.o 
ccache /usr/bin/clang++ -DIREE_ALLOCATOR_SYSTEM_CTL=iree_allocator_libc_ctl -I/home//workspace/experimental/iree -I/home//workspace/experimental/iree-build -I/home//workspace/experimental/iree/third_party/llvm-project/llvm/include -I/home//workspace/experimental/iree-build/llvm-project/include -I/home//workspace/experimental/iree/third_party/llvm-project/mlir/include -I/home//workspace/experimental/iree-build/llvm-project/tools/mlir/include -I/home//workspace/experimental/iree/third_party/llvm-project/lld/include -I/home//workspace/experimental/iree-build/llvm-project/tools/lld/include -I/home//workspace/experimental/iree/compiler/src -I/home//workspace/experimental/iree-build/compiler/src -I/home//workspace/experimental/iree/runtime/src -I/home//workspace/experimental/iree-build/runtime/src -I/home//workspace/experimental/iree-build/runtime/src/iree/base/internal/flatcc -isystem /home//workspace/experimental/iree/third_party/flatcc/include -O2 -g   -gsplit-dwarf -ggnu-pubnames -fPIC -fvisibility=hidden -fno-rtti -fno-exceptions -Wall -Wno-error=deprecated-declarations -Wno-extern-c-compat -Wno-invalid-offsetof -Wno-unused-const-variable -Wno-unused-function -Wno-unused-private-field -Wctad-maybe-unsupported -Wfloat-overflow-conversion -Wfloat-zero-conversion -Wfor-loop-analysis -Wformat-security -Wgnu-redeclared-enum -Wimplicit-fallthrough -Winfinite-recursion -Wliteral-conversion -Wlogical-op-parentheses -Wnon-virtual-dtor -Woverloaded-virtual -Wpointer-arith -Wself-assign -Wstring-conversion -Wtautological-overlap-compare -Wthread-safety -Wthread-safety-beta -Wunused-comparison -Wvla -fno-omit-frame-pointer -fmacro-prefix-map=/home//workspace/experimental/iree=iree -DIREE_THREADING_ENABLE=1 -pthread -I/home//workspace/experimental/iree/third_party/flatcc/include/ -I/home//workspace/experimental/iree/third_party/flatcc/include/flatcc/reflection/ -std=gnu++17 -MD -MT compiler/src/iree/compiler/Dialect/Util/Transforms/CMakeFiles/iree_compiler_Dialect_Util_Transforms_Transforms.objects.dir/HoistIntoGlobals.cpp.o -MF compiler/src/iree/compiler/Dialect/Util/Transforms/CMakeFiles/iree_compiler_Dialect_Util_Transforms_Transforms.objects.dir/HoistIntoGlobals.cpp.o.d -o compiler/src/iree/compiler/Dialect/Util/Transforms/CMakeFiles/iree_compiler_Dialect_Util_Transforms_Transforms.objects.dir/HoistIntoGlobals.cpp.o -c /home//workspace/experimental/iree/compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp
In file included from /home//workspace/experimental/iree/compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp:7:
In file included from /home//workspace/experimental/iree/compiler/src/iree/compiler/Dialect/Util/Analysis/Constant/ConstExpr.h:12:
In file included from /home//workspace/experimental/iree/third_party/llvm-project/llvm/include/llvm/ADT/DenseMap.h:20:
In file included from /home//workspace/experimental/iree/third_party/llvm-project/llvm/include/llvm/ADT/STLExtras.h:37:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/memory:80:
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/unique_ptr.h:857:34: error: constructor inherited by 'HoistIntoGlobalsPass' from base class 'HoistIntoGlobalsPassBase<mlir::iree_compiler::IREE::Util::(anonymous namespace)::HoistIntoGlobalsPass>' is implicitly deleted
    { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
                                 ^
/home//workspace/experimental/iree-build/compiler/src/iree/compiler/Dialect/Util/Transforms/Passes.h.inc:813:17: note: in instantiation of function template specialization 'std::make_unique<mlir::iree_compiler::IREE::Util::(anonymous namespace)::HoistIntoGlobalsPass>' requested here
    return std::make_unique<DerivedT>();
                ^
/home//workspace/experimental/iree-build/compiler/src/iree/compiler/Dialect/Util/Transforms/Passes.h.inc:751:33: note: in instantiation of member function 'mlir::iree_compiler::IREE::Util::impl::createHoistIntoGlobalsPass' requested here
  std::unique_ptr<::mlir::Pass> createHoistIntoGlobalsPass();
                                ^
/home//workspace/experimental/iree/compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp:355:7: note: constructor inherited by 'HoistIntoGlobalsPass' is implicitly deleted because field 'registerDependentDialectsFn' of const-qualified type 'const std::optional<ExprHoistingOptions::RegisterDialectsFn>' (aka 'const optional<function<void (mlir::DialectRegistry &)> >') would not be initialized
      registerDependentDialectsFn;
      ^
In file included from /home//workspace/experimental/iree/compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp:7:
In file included from /home//workspace/experimental/iree/compiler/src/iree/compiler/Dialect/Util/Analysis/Constant/ConstExpr.h:12:
In file included from /home//workspace/experimental/iree/third_party/llvm-project/llvm/include/llvm/ADT/DenseMap.h:20:
In file included from /home//workspace/experimental/iree/third_party/llvm-project/llvm/include/llvm/ADT/STLExtras.h:37:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/memory:80:
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/unique_ptr.h:857:34: error: constructor inherited by 'HoistIntoGlobalsPass' from base class 'HoistIntoGlobalsPassBase<mlir::iree_compiler::IREE::Util::(anonymous namespace)::HoistIntoGlobalsPass>' is implicitly deleted
    { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
                                 ^
/home//workspace/experimental/iree-build/compiler/src/iree/compiler/Dialect/Util/Transforms/Passes.h.inc:817:17: note: in instantiation of function template specialization 'std::make_unique<mlir::iree_compiler::IREE::Util::(anonymous namespace)::HoistIntoGlobalsPass, mlir::iree_compiler::IREE::Util::HoistIntoGlobalsPassOptions>' requested here
    return std::make_unique<DerivedT>(std::move(options));
                ^
/home//workspace/experimental/iree-build/compiler/src/iree/compiler/Dialect/Util/Transforms/Passes.h.inc:755:33: note: in instantiation of member function 'mlir::iree_compiler::IREE::Util::impl::createHoistIntoGlobalsPass' requested here
  std::unique_ptr<::mlir::Pass> createHoistIntoGlobalsPass(HoistIntoGlobalsPassOptions options);
                                ^
/home//workspace/experimental/iree/compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp:355:7: note: constructor inherited by 'HoistIntoGlobalsPass' is implicitly deleted because field 'registerDependentDialectsFn' of const-qualified type 'const std::optional<ExprHoistingOptions::RegisterDialectsFn>' (aka 'const optional<function<void (mlir::DialectRegistry &)> >') would not be initialized
      registerDependentDialectsFn;
      ^
2 errors generated.
```

### Steps to reproduce your issue

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


### What component(s) does this issue relate to?

_No response_

### Version information

_No response_

### Additional context

_No response_

--- a/compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp
+++ b/compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp
@@ -390,7 +390,7 @@ public:
 
 private:
   const std::optional<ExprHoistingOptions::RegisterDialectsFn>
-      registerDependentDialectsFn;
+      registerDependentDialectsFn = std::nullopt;
 };
 
 } // namespace

